libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/NavigationSession.kt

```diff
@@ -71,10 +69,24 @@ internal class NavigationSession : RoutesObserver, TripSessionStateObserver {
             TripSessionState.STOPPED -> false
         }
     }
+}
 
-    internal enum class State {
-        IDLE,
-        FREE_DRIVE,
-        ACTIVE_GUIDANCE
-    }
+/**
+ * Contains the various states that can occur during a navigation.
+ */
+enum class NavigationSessionState {
+    /**
+     * Idle state
+     */
+    IDLE,
+
+    /**
+     * Free Drive state
+     */
+    FREE_DRIVE,
+
+    /**
+     * Active Guidance state
+     */
+    ACTIVE_GUIDANCE
```

Comment:
We should be more specific here and describe the "requirements" for entering or leaving any of these states. A diagram would be great as well.